File: Selnium script for testing/Script with Python/test_scenario_7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

def execute_test_7():
    """
    Dummy test run for scenario 7
    """
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    driver.get("https://dummy.example.com/page_7")

    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.CSS_SELECTOR, ".promo-code").send_keys("test_data_237")
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.XPATH, "//button[@type='submit']").click()
    driver.find_element(By.LINK_TEXT, "Log Out").click()

    driver.quit()

if __name__ == "__main__":
    execute_test_7()

File: Selnium script for testing/Script with Java/src/test/java/testscenario/TestScenario7.java
package testscenario;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import java.time.Duration;

public class TestScenario7 {
    public static void executeTest7() {
        /*
         * Dummy test run for scenario 7
         */
        WebDriver driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get("https://dummy.example.com/page_7");

        try {
            driver.findElement(By.cssSelector(".promo-code")).click();
            if (!driver.getTitle().contains("Dashboard")) {
                System.out.println("Dashboard not in title");
            }
            driver.findElement(By.xpath("//button[@type='submit']")).click();
            driver.findElement(By.className("nav-item")).click();
            Thread.sleep(2000);
            driver.findElement(By.className("nav-item")).sendKeys("test_data_362");
            if (!driver.findElement(By.id("user_name")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.linkText("Log Out")).sendKeys("test_data_908");
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
    
    public static void main(String[] args) {
        executeTest7();
    }
}
